lib/ansible/module_utils/compat/importlib.py

```diff
@@ -3,14 +3,24 @@
 
 from __future__ import annotations
 
-import sys
+from ansible.module_utils.common.warnings import deprecate
 
-try:
-    from importlib import import_module  # pylint: disable=unused-import
-except ImportError:
-    # importlib.import_module returns the tail
-    # whereas __import__ returns the head
-    # compat to work like importlib.import_module
-    def import_module(name):  # type: ignore[misc]
-        __import__(name)
-        return sys.modules[name]
+
+def __getattr__(importable_name):
```

Comment:
@webknjaz Including a deprecation warning for this is a bit premature. Collections will be unable to eliminate the warning without breaking support for older Python versions supported by older versions of ansible-core which are still supported.

Instead of adding a run-time deprecation warning, use a `# deprecated:` comment to indicate when a deprecation warning shoud be added.

Reply:
Actually, this one should be OK to deprecate now. I'd forgotten this function was included in Python 2.7, which means any collection supporting ansible-core 2.13+ would have no issues with this deprecation.